src/main/java/it/pagopa/ecommerce/commons/queues/QueueEvent.java

```diff
@@ -0,0 +1,31 @@
+package it.pagopa.ecommerce.commons.queues;
+
+import it.pagopa.ecommerce.commons.documents.v1.TransactionEvent;
```

Comment:
Since this class refers to v1 TransactionEvent maybe this can be moved inside v1 subpackage in order to support future versioning. What do you think? Or a base TransactionEvent structure can be extracted from versioning package system, maybe one with only informations that will be shared with all future version events such as transactionId, creation date from which all future version will inherit.

Reply:
The better solution would be to extract a base `TransactionEvent` that is guaranteed to never change.
However this type bound is only for a bit of ergonomics (especially in tests) and not for actual type-safety:
 1. `QueueEvent` is meant as a simple container wrapping data + tracing information
 2. The only producer of `QueueEvent`s is `QueueAsyncClient`, the bound can be pushed there without impacts

I am not totally convinced on moving this under `v1`, as there would be no advantage to having a `v2.QueueEvent` (unless other tracing information become necessary, but putting such metadata in-band should be a last resort measure)

Reply:
Created new unversioned base event in #95 🙂